src/rt/lifetime.d

```diff
@@ -316,7 +313,7 @@ bool __setArrayAllocLength(ref BlkInfo info, size_t newlength, bool isshared, co
             *typeInfo = cast() tinext;
         }
     }
-    else if (info.size < PAGESIZE)
+    else if (info.size < pageSize)
```

Comment:
This should be the same size as the "large object" granularity in the GC (extending is expected to be supported for larger blocks), and must not exceed 64kB, as the size saved as the length of the allocated array has to fit into 2 bytes.